noma/bitcoind.py

```diff
@@ -186,23 +189,29 @@ def create():
 def set_prune(prune_target, config_path=""):
     """Set bitcoind prune target, minimum 550"""
     if not config_path:
-        config_path = "/media/archive/archive/bitcoin/bitcoin.conf"
-    set_kv("prune", prune_target, config_path)
+        config_path = cfg.BITCOIN_CONF
+    set_ini_kv("prune", prune_target, config_path, None)
 
 
-def set_rpcauth(config_path):
-    """Write new rpc auth to bitcoind and lnd config"""
-    import noma.lnd
+def set_rpcauth(bitcoin_conf=str(cfg.BITCOIN_CONF), invoicer_conf=str(cfg.INVOICER_CONF)):
+    """Write rpcauth to bitcoind and invoicer configs"""
+    auth_value, password = generate_rpcauth("lncm")
 
-    # TODO: Generate usernames too
-    if not config_path:
-        config_path = "/media/archive/archive/bitcoin/bitcoin.conf"
-    if pathlib.Path(config_path).is_file():
-        auth_value, password = generate_rpcauth("lncm")
-        set_kv("rpcauth", auth_value, config_path)
-        noma.lnd.set_bitcoind(password)
-    else:
-        create()
+    try:
+        rpcauth_val = get_ini_kv(bitcoin_conf, "rpcauth", None)
+    except IOError:
+        print("❌ Error: bitcoin config file not found")
+        raise
+    except KeyError:
+        print("rpcauth unset")
+
+    if rpcauth_val == '': # only change if value is unset
```

Comment:
I think it's better when "the happy path" is left most ;).

So flipping this `if` might improve clarity.

```python
if some-error:
	print
	return

back to happy path ^o^
```